app/webpacker/controllers/flatpickr_controller.js

```diff
@@ -60,6 +60,14 @@ export default class extends Flatpickr {
 
   open() {
     this.fp.element.dispatchEvent(new Event("focus"));
+    if (!this.defaultDateValue) return;
+    if (this.fp.selectedDates.length) return;
+
+    if (this.defaultDateValue === "startOfDay") {
+      this.fp.setDate(moment().startOf("day").format());
+    } else if (this.defaultDateValue === "endOfDay") {
+      this.fp.setDate(moment().add(1, "days").startOf("day").format());
```

Comment:
I'm wondering when you didn't use `moment().endOf('day')`?

https://momentjscom.readthedocs.io/en/latest/moment/03-manipulating/04-end-of/

Reply:
Hi @jibees 
I didn't use that function because the form ignores the seconds part from the date.
So, the report will not load the data between [23:59:00 ~> 23:59:59]
![image](https://user-images.githubusercontent.com/16113997/204245445-a8fdb9d4-4b48-4988-92ce-77b72eb0c265.png)
What do you think?

Reply:
That's a very edge case which you took into account, thanks!
I'd vote for a small comment in the code that explain that. What do you think?